internal/transports/p2p/peer/checkpoint.go

```diff
@@ -0,0 +1,139 @@
+package peer
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/bitcoin-sv/block-headers-service/domains"
+	"github.com/bitcoin-sv/block-headers-service/internal/chaincfg"
+	"github.com/bitcoin-sv/block-headers-service/internal/chaincfg/chainhash"
+	"github.com/rs/zerolog"
+)
+
+type checkpoint struct {
+	checkpoint      *chaincfg.Checkpoint
```

Comment:
Maybe we could consider renaming `checkpoint` -> `currentCheckpoint`? Because I'm constantly confusing it with `checkpoints` 😄